lib/commands/record-screen.js

```diff
@@ -0,0 +1,339 @@
+        }
+      });
+
+      this._process.proc.stdin.write('q');
+      this._process.proc.stdin.end();
+    });
+  }
+}
+
+
+/**
+ * @typedef {Object} StartRecordingOptions
+ *
+ * @property {?string} videoFilter - The video filter spec to apply for ffmpeg.
+ * See https://trac.ffmpeg.org/wiki/FilteringGuide for more details on the possible values.
+ * Example: Set it to `scale=ifnot(gte(iw\,1024)\,iw\,1024):-2` in order to limit the video width
+ * to 1024px. The height will be adjusted automatically to match the actual ratio.
+ * @property {number|string} fps [15] - The count of frames per second in the resulting video.
+ * The greater fps it has the bigger file size is.
+ * @property {string} preset [veryfast] - One of the supported encoding presets. Possible values are:
+ * - ultrafast
+ * - superfast
+ * - veryfast
+ * - faster
+ * - fast
+ * - medium
+ * - slow
+ * - slower
+ * - veryslow
+ * A preset is a collection of options that will provide a certain encoding speed to compression ratio.
+ * A slower preset will provide better compression (compression is quality per filesize).
+ * This means that, for example, if you target a certain file size or constant bit rate, you will achieve better
+ * quality with a slower preset. Read https://trac.ffmpeg.org/wiki/Encode/H.264 for more details.
+ * @property {boolean} captureCursor [false] - Whether to capture the mouse cursor while recording
+ * the screen
+ * @property {boolean} captureClicks [false] - Whether to capture mouse clicks while recording the
+ * screen
+ * @property {?string} audioInput - If set then the given audio input will be used to record the computer audio
+ * along with the desktop video. The list of available devices could be retrieved using
+ * `ffmpeg -list_devices true -f dshow -i dummy` command.
+ * @property {string|number} timeLimit [600] - The maximum recording time, in seconds. The default
+ * value is 600 seconds (10 minutes).
+ * @property {boolean} forceRestart [true] - Whether to ignore the call if a screen recording is currently running
+ * (`false`) or to start a new recording immediately and terminate the existing one if running (`true`).
+ */
+
+/**
+ * Record the display in background while the automated test is running.
+ * This method requires FFMPEG (https://www.ffmpeg.org/download.html) to be installed
+ * and present in PATH.
+ * The resulting video uses H264 codec and is ready to be played by media players built-in into web browsers.
+ *
+ * @param {?StartRecordingOptions} options - The available options.
+ * @throws {Error} If screen recording has failed to start or is not supported on the device under test.
+ */
+commands.startRecordingScreen = async function startRecordingScreen (options = {}) {
+  const {
+    timeLimit,
+    videoFilter,
+    fps,
+    preset,
+    captureCursor,
+    captureClicks,
+    audioInput,
+    forceRestart = true,
+  } = options;
+  if (this._screenRecorder) {
+    if (this._screenRecorder.isRunning()) {
```

Comment:
These two can just be 
```js
if (this._screenRecorder?.isRunning()) {
  // ...
}
this._screenRecorder = null;
```
The last line isn't needed, really, either, since it is set to the `new ScreenRecorder()` a couple of lines later.

Reply:
changed